File: hanoi.cpp
#include <iostream>

void hanoi(char A, char B, char C, int pans)
{
    if(pans == 0)
        return;

    hanoi(A,C,B,pans-1);
    std::cout<<"move No."<<pans<<" from "<<A<<" to "<<C<<std::endl;
    hanoi(B,A,C,pans-1);
}

int main()
{
    hanoi('A','B','C',3);
    return 0;
}

File: hanoi.java
public class hanoi
{
    public static void hanoi_(char A, char B, char C, int pans)
    {
        if(pans == 0)
            return;
        
            hanoi_(A,C,B,pans-1);
            System.out.printf("move No.%d from %c to %c\n", pans, A, C);
            hanoi_(B,A,C,pans-1);
    }

    public static void main(String[] args)
    {
        hanoi_('A','B','C',3);
    }
}